FirebaseSessions/Sources/SessionSampler.swift

```diff
@@ -0,0 +1,43 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+protocol SessionSamplerProtocol {
+  /// Sampling rate that has to be applied across sessions.
+  /// Ranges from 0 to 1 in Double.
+  var sessionSamplingRate: Double { get set }
```

Comment:
Does this need to be exposed to clients of types conforming to `SessionSamplerProtocol`? My hunch is no because a `SessionSamplerProtocol` should tell me whether or not to send an event– the sampling rate is an implementation detail.

Reply:
It is as per the design.

SessionSampler decides if an event needs to be sampled based on 2 factors - Event, Sampling rate.

Event is passed as an input and Sampling Rate is configurable (since that is an external factor that determines it - Configuration System).

So it is per the design to make it configurable.